pkg/catalog/helm/helm.go

```diff
@@ -239,6 +239,41 @@ func (h *Helm) fetchTgz(helmURL string) ([]v32.File, error) {
 	return files, nil
 }
 
+func (h *Helm) fetchTgz(helmURL string) ([]v32.File, error) {
+	var files []v32.File
+	logrus.Debugf("Helm fetching file %s", helmURL)
+
+	if h.IsLocalHelmRepo {
+		tgzPath := filepath.Join(h.LocalPath, helmURL)
+		r, err := os.Open(tgzPath)
```

Comment:
On first glance I don't like the idea of adding this into the helm package for rancher itself, opening files on the local cluster seems like it could be security hole. I haven't looked to deep into this but how hard would it be to keep all this logic in the separate image go program ? 

Can you just open up all the tgz files that exist in the assets and read them without caring about the index.yaml ? Or do you need to know sub-charts and stuff.

Reply:
@Caleb The index.yaml is used to figure out the latest version of a chart and the location of its tgz. 
What's happening here is we open the tgz to provide `readTgz()` with a reader so that it can be extracted to a separate directory. Then we update the version `version.Dir` with the path to that directory which is used in the image package to pick images from all values.yaml.
I will need to investigate a bit more, but I think we can move this logic to image. Although this will result in duplicated code for reading and extracting a tgz to a separate folder.

Reply:
I think duplication would be fine, we don't have to recreate whole helm dir package, just have to make it work for this use-case.
@StrongMonkey do you have opinion here ?

Reply:
@cbron I feel like it is ok, in helm we store these content in local disk so that we don't have to re-fetch it from internet. Besides that this code will only be called in image gen code.